src/reset-password/service.ts

```diff
@@ -25,3 +28,28 @@ export async function sendResetPasswordEmail(user: User) {
 
     nodemailer.getTestMessageUrl(info);
 }
+
+export async function createToken(user: User): Promise<string> {
+    const token = crypto.randomBytes(32).toString('base64url').replace('=', '');
+    const query = 'INSERT INTO reset_password_token SET user_id = ?, token = ?, created_at = UNIX_TIMESTAMP()';
+    await database.query(query, [
+        user.id,
+        await bcrypt.hash(token, 12),
+    ]);
+    return token;
+}
+
+export async function validateToken(user: User, token: string): Promise<boolean> {
```

Comment:
This function should prevent a token from being uses that older than a set amount of time. For example, 2 hours. It would be nice if a specific error would be thrown in this case.